### PR TITLE
CloudWatch DescribeLogGroups

### PR DESCRIPTION
**Pull request by aporubov**
This changeset implements [CloudWatch DescribeLogGroups](http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_DescribeLogGroups.html) API call.

@motobob please take a look

### CODE REVIEW

**Comment by motobob** [comment]
@aporubov https://github.com/erlcloud/erlcloud/blob/master/src/erlcloud_mon.erl is the cloudwatch file. could you please amend the PR

**Comment by aporubov** [comment]
@motobob please see [Amazon CloudWatch Documentation](https://aws.amazon.com/ru/documentation/cloudwatch/).

`erlcloud_mon` and `erlcloud_cloudwatch` modules implement bindings to two separate [CloudWatch](http://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/WhatIsCloudWatch.html) and [CloudWatch Logs](http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/WhatIsCloudWatchLogs.html) Amazon WebServices (there is also [CloudWatch Events](http://docs.aws.amazon.com/AmazonCloudWatch/latest/events/WhatIsCloudWatchEvents.html) service). Consequently, these two services expose separate APIs, please see respective [CloudWatch API Reference](http://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/Welcome.html) and [CloudWatch Logs API Reference](http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/Welcome.html). Thus, I don't think these two bindings should be mixed in a single module (for variety of reasons, e.g. CloudWatch is based on XML while CloudWatch Logs is JSON-based). I can rename `erlcloud_cloudwatch` to `erlcloud_cloudwatch_logs` for clarity.

File: src/erlcloud_cloudwatch.erl [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    }.
+
+
+%%==============================================================================
+%% API
+%%==============================================================================
+
+
+-spec describe_log_groups() -> result_paged(log_group()).
+describe_log_groups() ->
```
Comment (motobob) [inline src/erlcloud_cloudwatch.erl]: please add a link to AWS docs for every API

File: src/erlcloud_cloudwatch.erl [hunk 2]
```diff
@@ -0,0 +1,266 @@
+-module(erlcloud_cloudwatch).
```
Comment (motobob) [inline src/erlcloud_cloudwatch.erl]: yes, let's rename it to `erlcloud_cloudwatch_logs` or may be better `erlcloud_cw_logs`

File: src/erlcloud_cloudwatch.erl [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        {ok, Data} ->
+            LogGroups = proplists:get_value(<<"logGroups">>, Data, []),
+            NextToken = proplists:get_value(<<"nextToken">>, Data, undefined),
+            {ok, LogGroups, NextToken};
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+
+-spec describe_log_groups_all() -> result(log_group()).
```
Comment (motobob) [inline src/erlcloud_cloudwatch.erl]: let's not have `*_all` as we figured how to make it neat with ec2.

File: test/erlcloud_cloudwatch_tests.erl [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+%% An input test list provides a list of funs and the parameters that are
+%% expected to result.
+%%
+%% Output tests verify that the http response produces the correct return
+%% from the fun.
+%% An output test lists provides a list of response bodies and the
+%% expected return.
+
+
+%% The _rds_test macro provides line number annotation to a test,
```
Comment (motobob) [inline test/erlcloud_cloudwatch_tests.erl]: c/p

**Comment by aporubov** [comment]
@motobob all addressed

File: include/erlcloud_aws.hrl [hunk 5]
```diff
@@ -65,6 +65,9 @@
           cloudtrail_scheme="https://"::string(),
           cloudtrail_host="cloudtrail.amazonaws.com"::string(),
           cloudtrail_port=80::non_neg_integer(),
+          cloudwatch_scheme="https://"::string(),
```
Comment (motobob) [inline include/erlcloud_aws.hrl]: let's have it equal to module name

**Comment by motobob** [comment]
tnx. let's have a start for CW logs.

**Pull request merged**